astropy/wcs/wcs.py

```diff
@@ -2958,9 +2961,13 @@ def slice(self, view, numpy_order=True):
                     crp = ((crpix - iview.start - 1.)/iview.step
                            + 0.5 + 1./iview.step/2.)
                     wcs_new.wcs.crpix[wcs_index] = crp
+                    if wcs_new.sip is not None:
+                        sip_crpix[wcs_index] = crp
```

Comment:
I have to think about this. You can't simply change `sip.crpix`, th eorigin of the polynomial needs to be changed. The SIP correction, wcs_new.sip, on any given pixel must be equivalent to the correction of the same pixel in full frame.

Reply:
@nden That is the point of this: shift the origin of the polynomial (`crpix`) so that the value of the polynomial does not change for the same positions _relative_ to the `crpix`.

Reply:
That is, `SIP(new_x; new_crpix) = SIP(old_x, old_crpix)`. The test `all_pix2world(crpix) == crval` shows that this is correct.